File: VT13/VT13.py
n = int(input())
a = list(map(int, input().split()))

best_sum = -10**30
best_i = -1

for i in range(n):
    s = a[i] + a[(i + 1) % n]
    if s > best_sum or (s == best_sum and i > best_i):
        best_sum = s
        best_i = i

print(a[best_i], a[(best_i + 1) % n])

File: VT13/VT13.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    long long best_sum = LLONG_MIN;
    int best_i = -1;
    for (int i = 0; i < n; i++) {
        long long s = a[i] + a[(i + 1) % n];
        if (s > best_sum || (s == best_sum && i > best_i)) {
            best_sum = s;
            best_i = i;
        }
    }
    cout << a[best_i] << " " << a[(best_i + 1) % n] << endl;
    return 0;
}
